flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -141,10 +139,10 @@ public Catalog catalog() {
   }
 
   private Namespace toNamespace(String database) {
-    String[] namespace = new String[baseNamespace.length + 1];
-    System.arraycopy(baseNamespace, 0, namespace, 0, baseNamespace.length);
-    namespace[baseNamespace.length] = database;
-    return Namespace.of(namespace);
+    List<String> namespaces = Lists.newArrayListWithExpectedSize(baseNamespace.levels().length + 1);
+    Collections.addAll(namespaces, baseNamespace.levels());
+    Collections.addAll(namespaces, database);
```

Comment:
Do you need to change the logic here to use a `List`? What about changing `baseNamespace` references to `baseNamespace.levels()`? That would be fewer changes.